kiwi/container/docker.py

```diff
@@ -113,12 +116,24 @@ def create(self, filename):
             [container_dir, self.container_tag]
         )
 
-        Command.run(
-            ['umoci', 'init', '--layout', container_dir]
-        )
-        Command.run(
-            ['umoci', 'new', '--image', container_name]
-        )
+        if base_image:
+            uncompressor = Compress(base_image)
+            uncompressor.uncompress(True)
+            self.uncompressed_image = uncompressor.uncompressed_filename
+
+            Command.run([
+                'skopeo', 'copy',
+                'docker-archive:{0}'.format(self.uncompressed_image),
+                'oci:{0}'.format(container_name)
+            ])
```

Comment:
ok so how about the following:

- RootImportDocker::sync_data()

   takes care to embedd the derived image as uncompressed file and also creates a checksum
   from that file

- ContainerImageDocker::create()

   checks if the checksum matches the derived image file and only operates on it if this is the
   case. This will not prevent possible inconsistencies between a prepare and create call but
   it offers enough protection in my eyes. I understand that we want an atomic operation, thanks
   for the clarification

After all I think we should write some code comments on the workflow, why we keep the derived image in the tree, basically because I think this information can not be taken from just reading the code

Thoughts ?